src/Polysemy/HList.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE BlockArguments  #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Polysemy.HList
```

Comment:
Maybe `Polysemy.Several`? `HList` strikes me as an implementation detail. Happy to merge afterwards!

Reply:
Just pushed the change